l10n_eu_oss/wizard/l10n_eu_oss_wizard.py

```diff
@@ -0,0 +1,216 @@
+    )
+    second_superreduced_tax = fields.Many2one(
+        comodel_name="account.tax", string="Second Super Reduced Tax"
+    )
+
+    def _prepare_repartition_line_vals(self, original_rep_lines):
+        return [
+            (
+                0,
+                0,
+                {
+                    "factor_percent": line.factor_percent,
+                    "repartition_type": line.repartition_type,
+                    "account_id": line.repartition_type == "tax"
+                    and line.account_id.id
+                    or None,
+                    "company_id": line.company_id.id,
+                    "sequence": line.sequence,
+                },
+            )
+            for line in original_rep_lines
+        ]
+
+    def _prepare_tax_vals(self, country_id, tax_id, rate):
+        return {
+            "name": _("OSS for EU to {}: {}".format(country_id.name, rate)),
+            "amount": rate,
+            "invoice_repartition_line_ids": self._prepare_repartition_line_vals(
+                tax_id.invoice_repartition_line_ids
+            ),
+            "refund_repartition_line_ids": self._prepare_repartition_line_vals(
+                tax_id.refund_repartition_line_ids
+            ),
+            "type_tax_use": "sale",
+            "description": "EU-VAT-{}-{}".format(country_id.code, rate),
+            "oss_country_id": country_id.id,
+            "company_id": self.company_id.id,
+            "price_include": self.price_include_tax,
+            "sequence": 1000,
+        }
+
+    def generate_dict_taxes(self, selected_taxes, oss_rate_id):
+        dict_taxes = {}
+        # delete emptys values
+        oss_rate_id = [i for i in oss_rate_id if i != 0.0]
+        for idx, value in enumerate(selected_taxes):
+            dict_taxes[value] = oss_rate_id[
+                idx if idx < len(oss_rate_id) else len(oss_rate_id) - 1
+            ]
+        return dict_taxes
+
+    def _prepare_fiscal_position_vals(self, country, taxes_data):
+        fiscal_pos_name = _("Intra-EU B2C in %(country_name)s") % {
+            "country_name": country.name
+        }
+        fiscal_pos_name += " (EU-TAX-%s)" % country.code
+        return {
+            "name": fiscal_pos_name,
+            "company_id": self.company_id.id,
+            "vat_required": False,
+            "auto_apply": True,
+            "country_id": country.id,
+            "fiscal_position_type": "b2c",
+            "tax_ids": [(0, 0, tax_data) for tax_data in taxes_data],
+        }
+
+    def update_fpos(self, fpos_id, taxes_data):
+        fpos_id.mapped("tax_ids").filtered(
+            lambda x: x.tax_dest_id.oss_country_id
+        ).unlink()
+        fpos_id.write({"tax_ids": [(0, 0, tax_data) for tax_data in taxes_data]})
+
+    def generate_eu_oss_taxes(self):
+        oss_rate = self.env["oss.tax.rate"]
+        account_tax = self.env["account.tax"]
+        selected_taxes = []
+        fpos = self.env["account.fiscal.position"]
```

Comment:
For not conflicting with recordset, the convention is to name it with _obj suffix.